components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1395,9 +1394,16 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes,
 
     private void handleAndThrowClientExceptionForActionFailure(UserStoreException e) throws BadRequestException {
 
-        if (e instanceof UserStoreClientException && UserActionError.PRE_UPDATE_PASSWORD_ACTION_EXECUTION_FAILED
-                .equals(((UserStoreClientException) e).getErrorCode())) {
-            throw new BadRequestException(e.getMessage(), ResponseCodeConstants.INVALID_VALUE);
+        /* Note that UserActionExecutionClientException has been wrapped multiple times at this point and needs to be
+         unwrapped accordingly. Since there is a possibility of altering the error stack and not constructing the
+         failure response correctly, integration tests will be added to detect any regressions.
+         */
+        if (e instanceof UserStoreClientException && e.getCause() != null &&
```

Comment:
Better check the error code and if it's present we can find the cause. Else there could be a chance this breaks for other client exceptions which does not go to a depth like this level